test/tools/jsontests/BlockChainTests.cpp

```diff
@@ -416,19 +439,25 @@ json_spirit::mObject fillBCTest(json_spirit::mObject const& _input)
             cnote << testName + "block import throw an exception: " << diagnostic_information(_e);
             checkExpectedException(blObj, _e);
             eraseJsonSectionForInvalidBlock(blObj);
+            blockImportExceptionHappen = true;
         }
         catch (std::exception const& _e)
         {
             cnote << testName + "block import throw an exception: " << _e.what();
             cout << testName + "block import thrown std exeption\n";
             eraseJsonSectionForInvalidBlock(blObj);
+            blockImportExceptionHappen = true;
         }
         catch (...)
         {
             cout << testName + "block import thrown unknown exeption\n";
             eraseJsonSectionForInvalidBlock(blObj);
+            blockImportExceptionHappen = true;
         }
 
+        if (!_allowInvalidBlocks && blockImportExceptionHappen)
```

Comment:
What if invalid was expected, but exception was not thrown, will that be caught somewhere?

Reply:
yes it is being checked at function `checkExpectedException(blObj, _e);` in catch block above

Reply:
That looks like something else, because it's in `catch` block, so is checked only when excption is thrown.

I'm talking about the case when `_allowInvalidBlocks && !blockImportExceptionHappen`

Reply:
in case when `_allowInvalidBlocks && !blockImportExceptionHappen`  blocks without exception are allowed.

Reply:
Aren't those blocks invalid?
Do you mean that not all blocks in Invalid suite generate exceptions? Why are they in Invalid suite then?

Reply:
Sometimes for a block to be invalid chain must contain some valid blocks first